### DISCUSSION TITLE
SWG: What would cause it to go inactive in dashPanel?

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#845] [General]
Cell output goes to zero, SWG settings get greyed out, and I can't make any changes in dash.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/2a62e0da-76e4-44fb-ae20-46913100ea64)

This is been happening for a while.  I thought comms were messed up on my 7 year old IC40 but this is on a shiny new one I just installed.  Same thing.

Nice thing is I just have to remove the SWG in dash settings and add it back and it comes back and is responsive again.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/08b81c9a-a34d-4234-8228-7b563c387ebd)

Replay attached showing just that.

[replay (3).zip](https://github.com/tagyoureit/nodejs-poolController/files/12458707/replay.3.zip)

**Comment by rstrouse** [comment]
I assume that with this replay the chlorinator is enabled.  The condition you are seeing is when REM Chem is dosing acid it will turn off the chlorinator during the dosing period.  You can disable this by turning off pH priority.   In the poolState.json this condition will appear as enabled: false while the dosing is occuring.

**Reply by johnny2678** [reply]
yes, the SWG was enabled when the replay was taken.

I can turn off pH priority to test, but wondering if there's something more happening? My SWG can get locked out like this for hours at a time - sometimes more - but my acid dosing is set for 20 seconds max with a 10min mixing period when there is acid demand.

Shouldn't it only be locked out for that 20 seconds, and then resume SWG dosing when Acid Dosing is complete?

Also, I acid dose multiple times a day as demand requires, but this condition only happens very randomly.  Sometimes twice a day.  Sometimes not for 2 weeks.  Not every time there is acid dosing.

**Reply by johnny2678** [reply]
also I think the idea behind PH Priority was to avoid high concentrations of acid and Cl.  At least that's how the hold intelliPH system from pentair worked.  When it was time to dose acid, the cell would power down, or go into some kind of low power mode (all lights off), the acid pump would dose, and then the cell would come back on right where it left off. 

Not crazy about the idea of mixing chem in my lines so I'd rather keep PH priority on - just not sure why the salt cell stays locked out fo so long after acid dosing.

**Reply by rstrouse** [reply]
It should not and I use pH priority and have never had an issue.  I don't find myself manually dosing though I simply set it up and let the system do its thing.  May I recommend using volume dosing?  This is better on so many levels.  I have mine set for a max dose of 150mL and 1 hour of mix time.  While it rarely doses that amount it holds the pH level very well.  Typically within .05 of the setpoint now that the TA level of my pool is 50.  I keep the setpoint at 7.65.

**Comment by johnny2678** [comment]
My pH levels have been rock solid also.  Any variation can be attributed to a calibration or chem event (adding TA or CYA).
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/437a2afc-9c2f-465e-807f-ba7ad59c483f)

I never dose manually either, just dose by time instead of volume.

Here's my settings (after turning dose priority off).  
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/d8d3be10-2d03-44d0-b107-18572b01e61c)

Happy to try dosing by volume and turning dose priority on to see if the issue comes back?

**Comment by johnny2678** [comment]
going to ask for help one more time @rstrouse 

with my new SWG, I'm once again trying out the dynamic dosing feature.  Long story but my old SWG, which put out Cl for 7 years did NOT co-exist well with ORP readings when it was set to 100%.  Made the ORP readings nose dive and they never recovered.

Anyhoo... the new cell doesn't seem to have that affect on ORP so I wanted to try dynamic dosing.  I get through a couple of 15 minutes cycles of dosing/mixing and then it just shuts off.  It looks like it's in a state where it's not mixing, nor is it dosing.  And in my case ORP was not near the setpoint.  In the replay below, ORP setpoint was 690, and it stopped dosing around 650 and never came back on.  No status in the dashpanel, other than output is at 0%.  ~~Oh, and i have control of the SWG when this happens, which I thought was supposed to be grayed out when dynamic dosing is active.~~. edit: striking this for now as I'm not sure if this is related.

Any chance you could take a look?  It was doing just fine until around 1405 and then didn't come back.

It's a[ long replay](https://drive.google.com/file/d/15mTwXQEYEauXuRd3zwd3HrW6zP8kmvdV/view?usp=sharing) since I never know when it's going to shutdown. Hope the timestamp above helps.

**Reply by rstrouse** [reply]
Looking at your dose history I see it still dosing as of 15:15 and the setpoints are at 100%.

**Reply by johnny2678** [reply]
just happened again at 1911 after I reset the system and it dosed for 1-2 cycles.  Not dosing or mixing or mixing, but dynamic dosing still active.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/7d076c74-21d5-4737-86de-590bbe6882ee)

replay:
[replay (6).zip](https://github.com/tagyoureit/nodejs-poolController/files/12529031/replay.6.zip)

**Reply by johnny2678** [reply]
Problem is, it will sit at 0% output for hours until I realize it and reset and my FCl will dip.

**Reply by johnny2678** [reply]
> Looking at your dose history I see it still dosing as of 15:15 and the setpoints are at 100%.

according to my currentOutput readings, it was putting out 0% from 1404 to 1715, when I reset.

Then I had a weird cycle where dynamic dosing was turned on, but I had manual control of the SWG (the orange).  Then I got it going again and it cycled a few times and went back off.  Still off now.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/d9bd902f-56c2-4589-9957-38fe60e40b14)

**Reply by johnny2678** [reply]
another replay where dynamic dosing just quits.  This time around 1726.  I was out of the house so I didn't notice it and it ran at 0% fro two hours.  Restarted njsPC and it started dosing at 100% again.
[replay (5).zip](https://github.com/tagyoureit/nodejs-poolController/files/12543238/replay.5.zip)

edit: fixed replay
